samples/adc/multi_adc.cpp

```diff
@@ -17,23 +18,30 @@ int main() {
 
     io::UART& uart = io::getUART<io::Pin::UART_TX, io::Pin::UART_RX>(9600);
 
+    // Set up the logger to catch errors in ADC creation
+    core::log::LOGGER.setUART(&uart);
+    core::log::LOGGER.setLogLevel(core::log::Logger::LogLevel::INFO);
+
     uart.printf("Starting ADC test\r\n");
 
     time::wait(500);
 
-    io::ADC& adc0 = io::getADC<io::Pin::PA_0>();
-    io::ADC& adc1 = io::getADC<io::Pin::PA_1>();
+    io::ADC& adc0 = io::getADC<io::Pin::PA_0, io::ADCPeriph::ONE>();
+    io::ADC& adc1 = io::getADC<io::Pin::PC_4, io::ADCPeriph::ONE>();
 
     while (1) {
-        uart.printf("--------------------\r\n");
-        uart.printf("ADC0 : %d mV\r\n", static_cast<uint32_t>(adc0.read() * 1000));
-        uart.printf("ADC0: %d%%\r\n", static_cast<uint32_t>(adc0.readPercentage() * 100));
-        uart.printf("ADC0 raw: %d\r\n\r\n", adc0.readRaw());
-
-        uart.printf("ADC1 : %d mV\r\n", static_cast<uint32_t>(adc1.read() * 1000));
-        uart.printf("ADC1: %d%%\r\n", static_cast<uint32_t>(adc1.readPercentage() * 100));
-        uart.printf("ADC1 raw: %d\r\n", adc1.readRaw());
-        uart.printf("--------------------\r\n\r\n");
+        core::log::LOGGER.log(core::log::Logger::LogLevel::INFO, "--------------------");
+        core::log::LOGGER.log(
+            core::log::Logger::LogLevel::INFO, "ADC1 : %d mV", static_cast<uint32_t>(adc0.read() * 1000));
+        core::log::LOGGER.log(
+            core::log::Logger::LogLevel::INFO, "ADC1: %d%%", static_cast<uint32_t>(adc0.readPercentage() * 100));
+        core::log::LOGGER.log(core::log::Logger::LogLevel::INFO, "ADC1 raw: %d\r\n", adc0.readRaw());
+        core::log::LOGGER.log(
+            core::log::Logger::LogLevel::INFO, "ADC1 : %d mV", static_cast<uint32_t>(adc1.read() * 1000));
+        core::log::LOGGER.log(
+            core::log::Logger::LogLevel::INFO, "ADC1: %d%%", static_cast<uint32_t>(adc1.readPercentage() * 100));
+        core::log::LOGGER.log(core::log::Logger::LogLevel::INFO, "ADC1 raw: %d\r\n", adc1.readRaw());
```

Comment:
Should this extra new line be here? Same for the ADC above. Also both of your ADCs print with the same identifier, how are you suppose to know which is which?
```suggestion
        core::log::LOGGER.log(core::log::Logger::LogLevel::INFO, "ADC1 raw: %d", adc1.readRaw());
```